benchmark-framework/src/main/java/io/openmessaging/benchmark/Benchmark.java

```diff
@@ -159,52 +160,62 @@ public static void main(String[] args) throws Exception {
+                                DriverConfiguration.class);
+                        log.info("--------------- WORKLOAD : {} --- DRIVER : {}---------------", workload.name,
+                                driverConfiguration.name);
+
+                        // Stop any left over workload
+                        worker.stopAll();
 
-                    worker.initializeDriver(new File(driverConfig));
+                        worker.initializeDriver(new File(driverConfig));
 
-                    WorkloadGenerator generator = new WorkloadGenerator(driverConfiguration.name, workload, worker);
+                        WorkloadGenerator generator = new WorkloadGenerator(driverConfiguration.name, workload, worker);
 
-                    TestResult result = generator.run();
-                    result.beginTime = beginTime;
-                    result.endTime = dateFormat.format(new Date());
-                    result.version = arguments.serviceVersion;
+                        TestResult result = generator.run();
+                        result.beginTime = beginTime;
+                        result.endTime = dateFormat.format(new Date());
+                        result.version = arguments.serviceVersion;
 
-                    boolean useOutput = (arguments.output != null) && (arguments.output.length() > 0);
+                        boolean useOutput = (arguments.output != null) && (arguments.output.length() > 0);
 
-                    String fileName = useOutput? arguments.output: String.format("%s-%s-%s.json", workloadName,
-                    driverConfiguration.name, dateFormat.format(new Date()));
+                        String fileName = useOutput? arguments.output: String.format("%s-%s-%s.json", workloadName,
+                        driverConfiguration.name, dateFormat.format(new Date()));
 
-                    log.info("Writing test result into {}", fileName);
-                    writer.writeValue(new File(fileName), result);
+                        log.info("Writing test result into {}", fileName);
+                        writer.writeValue(new File(fileName), result);
 
-                    generator.close();
-                } catch (Exception e) {
-                    log.error("Failed to run the workload '{}' for driver '{}'", workload.name, driverConfig, e);
-                    // sometimes calling worker.stopAll hangs
-                    // using brute force to exit
-                    System.exit(1);
-                } finally {
-                    try {
-                        worker.stopAll();
-                    } catch (IOException e) {
+                        generator.close();
+                    } catch (Exception e) {
+                        log.error("Failed to run the workload '{}' for driver '{}'", workload.name, driverConfig, e);
+                        Throwables.throwIfUnchecked(e);
+                        throw new RuntimeException(e);
```

Comment:
This catch is here only to log now.

Note there are current/historical behaviors here:

1) Log when a workload fails, but continue. This is the upstream OMB behavior and the behavior before Denis's recent push. This was also the behavior of the original version of my change.
2) `System.exit()` immediately when a workload fails, no cleanup: this was how it worked in Denis' change.
3) Stop when a workload fails but do the normal cleanup, before calling `System.exit`. This is the behavior in this PR. Aligned with Denis' change, it means we don't continue on failure when the workers may be in an uncertain state.